imap/append.c

```diff
@@ -1063,7 +1079,7 @@ EXPORTED int append_fromstage_full(struct appendstate *as, struct body **body,
     r = msgrecord_get_fname(msgrec, &fname);
     if (r) goto out;
 
-    r = mailbox_copyfile(stagefile, fname, nolink);
+    r = mailbox_copyfile(linkfile ? linkfile : stagefile, fname, nolink);
     if (r) goto out;
```

Comment:
Unless I'm missing something, it looks like we are leaking `linkfile`.

Reply:
I've added a free()